From 3d1738f4c879419f743af55f285a272c16fd052d Mon Sep 17 00:00:00 2001
From: Liu Heng <liuheng@deepin.org>
Date: Tue, 10 Jan 2023 15:10:10 +0800
Subject: [PATCH 1/3] feat: add deepin auto build version fromat

when auto rebuild use deepin verion format(+deepin{x}) instead of obs release (+{reversion.buildtime})

log:
---
 build-recipe-dsc | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletions(-)

diff --git a/build-recipe-dsc b/build-recipe-dsc
index af3a502..d78e05e 100644
--- a/build-recipe-dsc
+++ b/build-recipe-dsc
@@ -73,6 +73,8 @@ recipe_prepare_dsc() {
     # remove rpm macros (everything after "%")
     # they are not evaluated by the Debian build process
     DEB_RELEASE=`sed 's/%.*$//' <<< $RELEASE`
+    DEB_BUILD_TIMES=`awk -F '.' '{print $2}' <<< $DEB_RELEASE`
+    DEB_RELEASE="rb`expr $DEB_BUILD_TIMES - 1`"
     OBS_DCH_RELEASE=""
 
 
@@ -97,6 +99,7 @@ recipe_prepare_dsc() {
                  RELEASEARGS="--release $DEB_RELEASE"
           if grep -Eq '^OBS-DCH-RELEASE: 1' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE \
-             && grep -Eq '^Version:' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE
+             && grep -Eq '^Version:' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE \
+             && [ $DEB_BUILD_TIMES -gt 1 ]
           then
             WSCHAR="`printf '\t '`"
             # https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-version
@@ -122,7 +125,7 @@ recipe_prepare_dsc() {
     chroot $BUILD_ROOT su -c "dpkg-source -x $DEB_SOURCEDIR/$DEB_DSCFILE $TOPDIR/BUILD" - $BUILD_USER
 
     # Alternative to debtransform: apply OBS release number if tag OBS-DCH-RELEASE is set.
-    if test -z "$DEB_TRANSFORM" && grep -Eq '^OBS-DCH-RELEASE: 1' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE; then
+    if test -z "$DEB_TRANSFORM" && grep -Eq '^OBS-DCH-RELEASE: 1' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE && [ $DEB_BUILD_TIMES -gt 1 ]; then
         OBS_DCH_RELEASE="+$DEB_RELEASE"
         chroot $BUILD_ROOT su -c /bin/sh <<EOF
 cd $TOPDIR/BUILD
 
-- 
2.20.1
